packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -470,6 +501,7 @@ export class TreeCheckout implements ITreeCheckoutFork {
 									);
 								}
 								this.disposeRevertible(revertible, revision);
+								this.disposeRevertibleRoots(revision);
```

Comment:
Have you found this to be necessary? I don't think any disposal is necessary. Everything that needs disposing will be disposed through trunk trimming (at least that's my expectation).

Reply:
I had thought that the policy was that repair data would be disposed as soon as the revertible holding it is disposed but if we can just let trunk trimming take care of it (which makes a lot more sense) then this def isn't needed

Reply:
There are two reasons why we may retain a piece of repair data:
1. The data was last touched by an edit that is still in the collab window (so other inbound edits may not be aware of it). These edits will not be trimmed.
2. That data was touched by an edit that the local client holds a revertible for. The approach of putting revertible commits on branches has the effect of preventing trunk trimming of those edits.

This means that we not only can rely on trunk trimming to tell us when we can GC repair data, we MUST NOT collect repair data in response to a revertible being disposed, because that revertible could be for an edit that is still in the collab window.